qiskit/opflow/gradients/gradient.py

```diff
@@ -41,20 +43,23 @@ class Gradient(GradientBase):
     # pylint: disable=signature-differs
     def convert(self,
                 operator: OperatorBase,
-                params: Union[ParameterVector, ParameterExpression, List[ParameterExpression]]
+                params: Optional[
+                    Union[ParameterVector, ParameterExpression, List[ParameterExpression]]] = None
                 ) -> OperatorBase:
         r"""
         Args:
             operator: The operator we are taking the gradient of.
-            params: params: The parameters we are taking the gradient with respect to.
+            params: The parameters we are taking the gradient with respect to. If not
+                explicitly passed, they are inferred from the operator and sorted by name.
```

Comment:
Ought there to be a test here that, if we default to taking the params from the Operator, that the operator indeed has some and if not raise an error?

Reply:
sure, let's raise the error here (in current implementation convert would return `ListOp` with empty `oplist`)

Reply:
Thinking more about it, maybe there should be standalone check if operator contains params? If we pass non-empty param list, but operator has no params, we should raise error as well.